lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).as_json
```

Comment:
The part of #954 that applies to my need (I just want `#as_json` or `#to_json`, hahahaha) looks great to me :+1: 

.. I don't have much to say on the rest of the changes - I'll let people more familiar with AMS's inner workings and roadmap provide more informed opinions on the rest.